gremlin-go/cucumber/featureSteps_test.go

```diff
@@ -0,0 +1,454 @@
+	}
+	return nil
+}
+
+func (tg *tinkerPopGraph) theResultShouldBe(characterizedAs string, table *godog.Table) error {
+	ordered := characterizedAs == "ordered"
+	switch characterizedAs {
+	case "empty":
+		if len(tg.result) != 0 {
+			return errors.New("actual result is not empty as expected")
+		}
+		return nil
+	case "ordered", "unordered", "of":
+		var expectedResult []interface{}
+		for idx, row := range table.Rows {
+			if idx == 0 {
+				// skip the header line
+				continue
+			}
+			val := parseValue(row.Cells[0].Value, tg.graphName)
+			v, ok := val.(*gremlingo.Path)
+			if ok {
+				// clear the labels since we don't define them in .feature files
+				v.Labels = []gremlingo.Set{}
+				val = v
+			}
+			expectedResult = append(expectedResult, val)
+		}
+		var actualResult []interface{}
+		for _, res := range tg.result {
+			actualResult = append(actualResult, res.GetInterface())
+		}
+		fmt.Println("EXPECTED", expectedResult)
+		fmt.Println("ACTUAL", actualResult)
+		if characterizedAs != "of" && len(actualResult) != len(expectedResult) {
+			err := fmt.Sprintf("actual result length %d does not equal to expected result length %d.", len(actualResult), len(expectedResult))
+			return errors.New(err)
+		}
+		if ordered {
+			for idx, res := range actualResult {
+				if !reflect.DeepEqual(expectedResult[idx], res) {
+					return errors.New("actual result is not ordered")
+				}
+			}
+		} else {
+			for _, res := range actualResult {
+				if !contains(expectedResult, res) {
+					return errors.New("actual result does not match expected result")
+				}
+			}
+		}
+		break
+	default:
+		return errors.New("scenario not supported")
+	}
+	return nil
+}
+
+func contains(list []interface{}, item interface{}) bool {
+	for _, v := range list {
+		if reflect.DeepEqual(v, item) {
+			return true
+		}
+	}
+	return false
+}
+
+func (tg *tinkerPopGraph) theResultShouldHaveACountOf(expectedCount int) error {
+	actualCount := len(tg.result)
+	if len(tg.result) != expectedCount {
+		err := fmt.Sprintf("result should return %d for count, but returned %d.", expectedCount, actualCount)
+		return errors.New(err)
+	}
+	return nil
+}
+
+func (tg *tinkerPopGraph) theTraversalOf(arg1 *godog.DocString) error {
+	traversal, err := GetTraversal(tg.scenario.Name, tg.g, tg.parameters)
+	if err != nil {
+		return err
+	}
+	tg.traversal = traversal
+	return nil
+}
+
+func (tg *tinkerPopGraph) usingTheParameterDefined(name string, params string) error {
+	if tg.graphName == "empty" {
+		tg.reloadEmptyData()
+	}
+	tg.parameters[name] = parseValue(strings.Replace(params, "\\\"", "\"", -1), tg.graphName)
+	return nil
+}
+
+func (tg *tinkerPopGraph) usingTheParameterOfP(paramName, pVal, stringVal string) error {
+	// TODO after fully implementing P class
+	// P class need to be able to take in interface{} for it's value arguments
+	//var p = reflect.ValueOf(gremlingo.P).MethodByName(pVal).Call(parseValue(stringVal, tg.graphName))
+	return godog.ErrPending
+}
+
+// safe?
```

Comment:
What is this comment asking?

Reply:
Just wondering if it is okay to create a global variable here.